boards: esp32c3_devkitc: Added board

raffarost opened the pull request:
Support for board esp32c3_devkitc

Commit:
boards: esp32c3_devkitc: Added support

Add support to esp32c3_devkitc-02

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

kartben commented:
it looks to me as the two commits should probably be two different PRs?

uLipe reviewed (changes requested):
As @kartben mentioned, there are two commits and one of them is unrelated to the PR.

Please split this PR into two separated one. Optionally you can open and merge a PR for the first patch, and make this one blocked by it.

Commit:
samples: esp32c3: Move config files from boards to socs

Move samples config files from 'boards' to 'socs' in order to
remove multiple files with the same configuration and render
available samples for new boards. Only changed sample files
which are not board or hardware specific.

Signed-off-by: Raffael Rostagno <raffael.rostagno@espressif.com>

raffarost commented:
Sorry I still didn't grasp the best practices. It was not thoughtful to include other boards in the PR.
However to me it makes sense to keep the second commit limited to esp32c3, as the idea is to render all samples moved from `boards` to `socs` available to the new board supported (esp32c3_devkitc). Is that okay?

kartben commented on boards/espressif/esp32c3_devkitc/Kconfig.defconfig:
```diff
@@ -0,0 +1,11 @@
+
+config HEAP_MEM_POOL_ADD_SIZE_BOARD
+	int
+	default 65535 if WIFI && BT
```
fwiw other esp32 boards have recently been updated to:

```suggestion
	default $(UINT16_MAX) if WIFI && BT
```

not sure this was fully intended though, as it looks to me as if this is a reasonably arbitrary figure that just happens to be equal to UINT16_MAX, not really something that's semantically *meant* to be UINT16_MAX

either way, it should probably be consistent across esp32 boards

marekmatej replied:
A more suitable macro here would be `KB(64)`   :)

sylvioalves replied:
@kartben Regarding the UINT16_MAX in those Kconfig, I think it is totally wrong. I approved that change though. I might submit a PR to restore it to `65535` or KB(64). This value has nothing to do with `UINT16_MAX` at all.

kartben replied:
@sylvioalves I agree it should be reverted -- and I should point out I also missed it during review :)

kartben commented:
@raffarost just stumbled upon this but it also needs `esp32c3_usb.conf` / `esp32c3_usb.overlay` files to cover the boards that are using the USB variant of the SoC

The below fails otherwise, until `tests/drivers/adc/adc_api/socs/esp32c3_usb.conf` / `tests/drivers/adc/adc_api/socs/esp32c3_usb.overlay` are added.

```
west build -p -b esp32c3_luatos_core/esp32c3/usb tests/drivers/adc/adc_api -T drivers.adc
```

Could you please send a follow-up PR?
